include/libdnf/transaction/comps_environment.hpp

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright (C) 2017-2020 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef LIBDNF_TRANSACTION_COMPS_ENVIRONMENT_HPP
+#define LIBDNF_TRANSACTION_COMPS_ENVIRONMENT_HPP
+
+
+#include "comps_group.hpp"
+
+#include <memory>
+#include <vector>
+
+
+namespace libdnf::transaction {
+
+
+class CompsEnvironmentGroup;
+class Transction;
+
+
+class CompsEnvironment : public TransactionItem {
+public:
+    explicit CompsEnvironment(Transaction & trans);
+
+    const std::string & get_environment_id() const noexcept { return environment_id; }
+    void set_environment_id(const std::string & value) { environment_id = value; }
```

Comment:
Why there is a setter in public API? Goal can use a private function as a friend.

Reply:
The same question is applicable for all setters.

Reply:
I'm not sure if the setters can be turned protected or private now.
They're needed in the tests and until the Goal can populate Transaction incl. transaction items,
I'm not sure if we can hide them.